packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/apis/helpers/validation.ts

```diff
@@ -100,10 +103,30 @@ export class ValidationHelper {
   private getTypeValidator(type: string): SavedObjectsTypeValidator {
     if (!this.typeValidatorMap[type]) {
       const savedObjectType = this.registry.getType(type);
+
+      const stackVersionSchemas =
+        typeof savedObjectType?.schemas === 'function'
+          ? savedObjectType.schemas()
+          : savedObjectType?.schemas ?? {};
+
+      const modelVersionCreateSchemas =
+        typeof savedObjectType?.modelVersions === 'function'
+          ? savedObjectType.modelVersions()
+          : savedObjectType?.modelVersions ?? {};
+
+      const combinedSchemas = { ...stackVersionSchemas };
+      Object.entries(modelVersionCreateSchemas).reduce((map, [key, modelVersion]) => {
+        const virtualVersion = modelVersionToVirtualVersion(key);
+        if (modelVersion.schemas?.create) {
+          combinedSchemas[virtualVersion] = modelVersion.schemas!.create!;
+        }
```

Comment:
NIT: `modelVersionToVirtualVersion` can be done within the `if` block to avoid calling it when not needed

```ts
if (modelVersion.schemas?.create) {
    const virtualVersion = modelVersionToVirtualVersion(key);
    combinedSchemas[virtualVersion] = modelVersion.schemas!.create!;
}
```

Reply:
done: https://github.com/elastic/kibana/pull/158527/files#diff-a6987e7e73107abdbc32cc43871a76dfec7cc99dfc2e9fc815cd338e4de0c457R119-R122